src/App.js

```diff
@@ -281,12 +283,36 @@ class App extends React.Component<PropsT, StateT> {
               this.recordNewSession(userId);
             }
           }}
+          onClickStartMamiferes={() => {
+            this.setState({
+              hasBeenWelcomed: true,
+              view: "mammiferes"
+            });
+            if (isRegistered && userId) {
+              this.recordNewSession(userId);
+            }
+          }}
+          onClickStartAdverbs={() => {
+            this.setState({
+              hasBeenWelcomed: true,
+              view: "adverbs"
+            });
+            if (isRegistered && userId) {
+              this.recordNewSession(userId);
+            }
+          }}
```

Comment:
Here you have one function repeated three times. In that cases it is good to extract it

```
const startCourse = courseType => {
    this.setState({  hasBeenWelcomed: true,  view: courseType});
￼    if (isRegistered && userId) {
￼       this.recordNewSession(userId);
￼    }
}


...
...
onClickStartAdverbs={() => startCourse("adverbs")}
...
...
```
Also right now the three views are called "training", "mammiferes" and "adverbs". "training" should be renamed "parallelogram".

Reply:
Also mammifère in English is mammal